app/models/thread.php

```diff
@@ -0,0 +1,120 @@
+<?php
+class Thread extends AppModel
+{
+    const MIN_STRING_LENGTH = 1;
+    const MAX_STRING_LENGTH = 80;
+
+    public $validation = array(
+       'title' => array(
+            'length' => array(
+               'validate_between', self::MIN_STRING_LENGTH, self::MAX_STRING_LENGTH,
+            ),
+        ),
+    );
+
+    public static function getAll($offset, $limit)
+    {
+        $threads = array();
+        $db = DB::conn();
+        $query = sprintf("SELECT * FROM thread LIMIT %d, %d", $offset, $limit);
+        $rows = $db->rows($query);
+        foreach ($rows as $row) {
+            $threads[] = new self($row);
+        }
+        return $threads;
+    }
+
+    public static function countAll()
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM thread");
+    }
+
+    public static function countAllComments($thread_id)
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ?", array($thread_id));
+    }
+
+    public static function get($id)
+    {
+        $db = DB::conn();
+        $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
+        if (!$row) {
+            throw new RecordNotFoundException('no record found');
+        }
+        return new self($row);
+    }
+
+    public function getComments($offset, $limit)
+    {
+        $comments = array();
+        $db = DB::conn();
+        $query = sprintf("SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC LIMIT %d, %d", $offset, $limit);
+        $rows = $db->rows($query, array($this->id));
+        foreach ($rows as $row) {
+            $row['username'] = $this->getUserName($row['user_id']);
+            $comments[] = new Comment($row);
+        }
+        return $comments;
```

Comment:
L51-59 should be in the comment model and this method should call from that model
ie.
```php
return Comment::getByThreadId($this->id, $offset, $limit);
```